svgpathtools/svg2paths.py

```diff
@@ -37,19 +43,21 @@ def ellipse2pathd(ellipse):
     return d
 
 
-def polyline2pathd(polyline_d):
+def polyline2pathd(polyline_d, is_polygon=False):
     """converts the string from a polyline points-attribute to a string for a
     Path object d-attribute"""
-    points = polyline_d.replace(', ', ',')
-    points = points.replace(' ,', ',')
-    points = points.split()
+    points = COORD_PAIR_TMPLT.findall(polyline_d)
+    closed = (float(points[0][0]) == float(points[-1][0]) and
+              float(points[0][1]) == float(points[-1][1]))
 
-    closed = points[0] == points[-1]
+    # The `parse_path` call ignores redundant 'z' (closure) commands
+    # e.g. `parse_path('M0 0L100 100Z') == parse_path('M0 0L100 100L0 0Z')`
+    # This check ensures that an n-point polygon is converted to an n-Line path.
+    if is_polygon and closed:
+        points.append(points[0])
 
-    d = 'M' + points.pop(0).replace(',', ' ')
-    for p in points:
-        d += 'L' + p.replace(',', ' ')
-    if closed:
+    d = 'M' + 'L'.join('{0} {1}'.format(x,y) for x,y in points)
+    if is_polygon or closed:
```

Comment:
The `or closed` seems a bit dangerous -- the "z" being redundant in this case.  Any argument for why to leave this?

Reply:
I did not change the logic of parent commit:
- in polygon2paths 'z' was appended in any case (along with appending of an additional segment),
- in polyline2paths 'z' was appended if the ends of the path were equal.
This condition remains that logic.

Reply:
OK, perfect! Thanks for the explanation :)